sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -1259,24 +1264,24 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
             StringBuilder payload = new StringBuilder();
 
             payload.append("resource=");
-            payload.append(URLEncoder.encode(resource, "UTF-8"));
+            payload.append(URLEncoder.encode(resource, StandardCharsets.UTF_8.name()));
             payload.append("&api-version=");
-            payload.append(URLEncoder.encode(endpointVersion, "UTF-8"));
+            payload.append(URLEncoder.encode(endpointVersion, StandardCharsets.UTF_8.name()));
             if (clientId != null) {
                 payload.append("&client_id=");
-                payload.append(URLEncoder.encode(clientId, "UTF-8"));
+                payload.append(URLEncoder.encode(clientId, StandardCharsets.UTF_8.name()));
             }
             if (resourceId != null) {
                 payload.append("&mi_res_id=");
-                payload.append(URLEncoder.encode(resourceId, "UTF-8"));
+                payload.append(URLEncoder.encode(resourceId, StandardCharsets.UTF_8.name()));
             }
             try {
                 URL url = new URL(String.format("%s?%s", endpoint, payload));
                 connection = (HttpURLConnection) url.openConnection();
 
                 connection.setRequestMethod("GET");
                 if (headerValue != null) {
-                    if (endpointVersion.equals(IDENTITY_ENDPOINT_VERSION)) {
+                    if (IDENTITY_ENDPOINT_VERSION.equals(endpointVersion)) {
```

Comment:
Have the constant validate the instance variable to prevent a possible NPE (don't believe it can occur in this case but it's just good practice)